### DISCUSSION TITLE
Saving settings

### DISCUSSION THREAD

**Discussion by wollew** [mriscoc/Ender3V2S1#1172] [General]
Hi everyone. A few months ago I bought a used Ender3v2 which was already flashed with the Professional firmware and this printer has mostly been running fine for me. I would now like to try Klipper (or even just update to the latest release of this firmware) but I would also like to keep the possibility to go back to the firmware that I know is working perfectly fine for me.  I am not too worried about being able to flash back and forth, what I currently don't know is which settings of my current firmware I somehow need to save and also how. So my question: is it good enough to just write down the output of the M503 gcode or do I somehow require something else?

**Comment by mriscoc** [comment]
Better: https://github.com/mriscoc/Ender3V2S1/wiki/Configuration-files

**Reply by wollew** [reply]
Thank you for your reply. So if I understand it right I was on the right track: use M503 to get the current configuration, save them to a config gcode file and in case I want to go back I use that to configure my printer.
